src/dc_msg.rs

```diff
@@ -269,87 +270,37 @@ pub unsafe fn dc_msg_empty(mut msg: *mut dc_msg_t) {
-    mut ret_msgtype: *mut Viewtype,
-    mut ret_mime: *mut *mut libc::c_char,
-) {
-    let mut suffix: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut dummy_msgtype = Viewtype::Unknown;
-    let mut dummy_buf: *mut libc::c_char = 0 as *mut libc::c_char;
-    if !pathNfilename.is_null() {
-        if ret_msgtype.is_null() {
-            ret_msgtype = &mut dummy_msgtype
-        }
-        if ret_mime.is_null() {
-            ret_mime = &mut dummy_buf
-        }
-        *ret_msgtype = Viewtype::Unknown;
-        *ret_mime = 0 as *mut libc::c_char;
-        suffix = dc_get_filesuffix_lc(pathNfilename);
-        if !suffix.is_null() {
-            if strcmp(suffix, b"mp3\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Audio;
-                *ret_mime = dc_strdup(b"audio/mpeg\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"aac\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Audio;
-                *ret_mime = dc_strdup(b"audio/aac\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"mp4\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Video;
-                *ret_mime = dc_strdup(b"video/mp4\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"jpg\x00" as *const u8 as *const libc::c_char) == 0i32
-                || strcmp(suffix, b"jpeg\x00" as *const u8 as *const libc::c_char) == 0i32
-            {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/jpeg\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"png\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/png\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"webp\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/webp\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"gif\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Gif;
-                *ret_mime = dc_strdup(b"image/gif\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"vcf\x00" as *const u8 as *const libc::c_char) == 0i32
-                || strcmp(suffix, b"vcard\x00" as *const u8 as *const libc::c_char) == 0i32
-            {
-                *ret_msgtype = Viewtype::File;
-                *ret_mime = dc_strdup(b"text/vcard\x00" as *const u8 as *const libc::c_char)
-            }
-        }
-    }
-    free(suffix as *mut libc::c_void);
-    free(dummy_buf as *mut libc::c_void);
+pub fn dc_msg_guess_msgtype_from_suffix(path: &Path) -> Option<(Viewtype, &str)> {
+    let known = hashmap! {
```

Comment:
given that this is static, I would prefer to use https://github.com/sfackler/rust-phf

Reply:
Possible. Requires unstable feature.

Reply:
if you use the git dependency it does not anymore: https://github.com/sfackler/rust-phf

Reply:
https://github.com/sfackler/rust-phf/pull/160

Reply:
Ok.